### PR TITLE
Fix SA1642 for Microsoft.PowerShell.Commands.Utility

### PR DESCRIPTION
**Pull request by xtqqczze**
Fix SA1642: Constructor summary documentation should begin with standard text for `Microsoft.PowerShell.Commands.Utility`.

https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1642.md

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs [hunk 1]
```diff
@@ -139,10 +139,12 @@ public override int GetHashCode()
     internal class ObjectCommandComparer : IComparer
     {
         /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectCommandComparer"/> class.
         /// Constructor that doesn't set any private field.
         /// Necessary because compareTo can compare two objects by calling
         /// ((ICompare)obj1).CompareTo(obj2) without using a key.
         /// </summary>
+        // TODO: Clarify summary text: "doesn't set any private field".
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs:147]: ?

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs [hunk 2]
```diff
@@ -139,10 +139,12 @@ public override int GetHashCode()
     internal class ObjectCommandComparer : IComparer
     {
         /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectCommandComparer"/> class.
         /// Constructor that doesn't set any private field.
```
Comment (xtqqczze) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs:143]: @iSazonov This text needs clarifying, the constructor is setting fields.

**Comment by iSazonov** [comment]
@xtqqczze Please see CI fails.

**Comment by iSazonov** [comment]
@xtqqczze Up.

**Comment by xtqqczze** [comment]
> @xtqqczze Please see CI fails.

I accidentally committed `dotnet_diagnostic.SA1642.severity = warning`.

**Review by iSazonov — APPROVED** [review]

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs (hunk 2 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs:143]: @iSazonov Should we update the comment in this PR?

Comment (iSazonov) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs:143]: I think the comment is correct but tricky - it means `((ICompare)obj1).CompareTo(obj2)` should works.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs (hunk 1 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs:147]: removed

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs (hunk 2 above)
Comment (xtqqczze) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/ObjectCommandComparer.cs:143]: OK, I have removed TODO.

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
I have removed TODO commit.

**Pull request merged**